schemas/context/experienceevent.example.1.json

```diff
@@ -5,26 +5,31 @@
     "xdm:code": "DataSourceIntegrationCode-123"
   },
   "xdm:timestamp": "2017-09-26T15:52:25+00:00",
-  "xdm:endUserIDs": {
-    "https://ns.adobe.com/experience/mcid": {
-      "xdm:id": "92312748749128",
-      "xdm:namespace": {
-        "xdm:code": "ECID"
+  "xdm:identityMap": {
+    "https://data.adobe.io/entities/namespace/4": [
+      {
+        "xdm:id": "92312748749128",
+        "xdm:namespace": {
+          "xdm:code": "ECID"
+        }
       }
-    },
-    "https://ns.adobe.com/experience/aaid": {
-      "xdm:id": "2394509340-30453470347",
-      "xdm:namespace": {
-        "xdm:code": "AVID"
+    ],
+    "https://data.adobe.io/entities/namespace/10": [
+      {
+        "xdm:id": "2394509340-30453470347",
+        "xdm:namespace": {
+          "xdm:code": "AVID"
```

Comment:
Is the code redundant? I take it this part is optional and all of xdm:namespace could be considered optional?

Reply:
If the intention of using an ID is to have a stable reference to the namespace and not have systems use the code in their implementations, then we should remove it. There's nothing stopping any implementation from assuming the code can be used for generating a fully qualified identity.

Reply:
This structure allows for specifying different namespace codes mapped to the same namespace key. Is that OK? If not, xdm:namespace should be outside the id object.

Reply:
Since the code can change, it makes sense to allow for the _current_ value to (optionally) be captured in the data.

Reply:
If we want both id and code how we are planning to get id at the time of ETL. Are we saying ETL ecosystem with integrate with Adobe Service to fetch id?

Reply:
Also, if keys are going to be represented as "https://data.adobe.io/entities/namespace/4" how are we use this in json pointer notation. REF-> https://tools.ietf.org/html/rfc6901

Reply:
@rkpandey the mechanism is that external callers use the integration code (as it is the user-facing "handle" for the namespace), and the system transforms to id for storage as appropriate.  The interface is the code, while the id is used internally.

There is no incompatibility between URIs and JSON Pointer.  JSON Pointer allows for escaping of reserved characters.